app/controllers/parties_controller.rb

```diff
@@ -0,0 +1,37 @@
+class PartiesController < ApplicationController
+  def show
+    @party = Party.find(params[:id])
+    current_user_guest_list_check(@party)
```

Comment:
Couple of things here. As this method calls a redirect, you might think the rest of the action won't take place, but it will. It would probably be better to have this check be explicit and ditch this method. So you'd do 
```
if current_user.on_guest_list?(party)
  @rsvp = Rsvp.new
  @location_query = @party.location.split(' ').join('+')
  @attendees = @party.users
  @restrictions = @party.combine_nonos    
else
  redirect_to user_path(current_user)
end
```

It might seem less D-R-Y but it's more readable and keeps the page flow in the controller action. In one sense this check is part  of the story of this controller action, so hiding it in a method call with side effects makes your application logic less clear.